invenio_records_lom/utils/vcard.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Graz University of Technology.
+#
+# invenio-records-lom is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Module with utilities for handling `vcard` version 4.0 (see `RFC6350 <https://www.rfc-editor.org/rfc/rfc6350>`_ for the spec).
+
+Usually, :py:obj:`make_lom_vcard` should be what you need.
+
+.. code-block:: python
+
+   # can be used directly, without any setup:
+   vcard = make_lom_vcard(**supported_vcard_properties)
+   # `vcard`'s type will be `str`
+
+To configure how VCards are made, create your own :py:class:`VCardMaker`,
+then use its :py:meth:`~VCardMaker.make_vcard` method:
+
+.. code-block:: python
+
+   my_vcard_maker = VCardMaker(**my_config)
+   vcard = my_vcard_maker.make_vcard(**supported_vcard_properties)
+   # `vcard`'s type will be type `str` or `bytes`, depending on `my_config`
+"""
+from __future__ import annotations  # allow `str | None` notation
+
+from collections.abc import Iterable
+from dataclasses import dataclass
+
+
+@dataclass
```

Comment:
question: Why do you use dataclass here? Is it only to not define the constructor for the 3 properties?

Reply:
`@dataclass` also adds some other dunder methods, which are recommended to be created anyway
(e.g. `__repr__`, `__eq__`, `__hash__`)

Reply:
yes, but why are they necessary in this context, if you only call `make_vcard` in the end?